internal/controllers/admins.go

```diff
@@ -44,75 +45,47 @@ func (s Server) UpdateUsages(ctx echo.Context) error {
 		err error
 		req UpdateUsagesReq
 	)
-
 	if err = ctx.Bind(&req); err != nil {
 		return ctx.JSON(http.StatusBadRequest, model.ErrorResponse(err.Error(), http.StatusBadRequest))
 	}
-
 	effectivedate, err := time.Parse("2006-01-02", req.EffectiveDate)
 	if err != nil {
 		return ctx.JSON(http.StatusBadRequest, model.ErrorResponse(err.Error(), http.StatusBadRequest))
 	}
-
+	var resourceType = model.ResourceType{Name: req.ResourceType}
+	s.GORMDB.Debug().Find(&resourceType)
+	resourceTypeID := *resourceType.ID
 	usageDetails := []model.Usage{}
-
-	if err = s.GORMDB.Debug().Raw(
-		`
-			SELECT usages.* 
-			FROM user_plans
-			JOIN usages ON user_plans.id=usages.user_plan_id
-			JOIN quotas ON user_plans.id=quotas.user_plan_id
-			JOIN resource_types ON resource_types.id=quotas.resource_type_id
-			JOIN users ON users.id = user_plans.user_id
-			WHERE user_plans.effective_start_date <= CURRENT_DATE
-			AND user_plans.effective_end_date >= CURRENT_DATE
-			AND users.user_name = ?
-			AND resource_types.name = ?
-		`,
-		req.UserName,
-		req.ResourceType,
-	).Scan(&usageDetails).Error; err != nil {
+	err = s.GORMDB.Debug().Table("user_plans").Select("usages.*").Joins("JOIN usages ON user_plans.id=usages.user_plan_id").Joins("JOIN resource_types ON resource_types.id=usages.resource_type_id").Joins("JOIN quota ON user_plans.id=quota.user_plan_id").Joins("JOIN users ON users.id = user_plans.user_id").Where("resource_types.name=? AND users.user_name=?", req.ResourceType, req.UserName).Scan(&usageDetails).Error
+	if err != nil {
 		return ctx.JSON(http.StatusInternalServerError, model.ErrorResponse(err.Error(), http.StatusInternalServerError))
 	}
 	for _, usagerec := range usageDetails {
 		usagerec.UpdatedAt = effectivedate
 		value := req.UsageAdjustmentValue
-		if req.UpdateType == "sub" {
+		if req.UpdateType == "SET" {
 			value = -1 * value
+		} else if req.UpdateType == "ADD" {
+			value = 1 * value
 		}
 		usagerec.Usage += value
```

Comment:
There should probably be constants for the available update types. Something like:

```
const (
	UpdateTypeSet = "SET"
	UpdateTypeAdd = "ADD"
)
```

Also this block doesn't quite implement the operations as they're defined. You might want to go with something like this instead:

```suggestion
		switch req.UpdateType {
		case UpdateTypeSet:
			usagerec.Usage = value
		case UpdateTypeAdd:
			usagerec.Usage += value
		default:
			msg := fmt.Sprintf("invalid update type: %s", req.UpdateType)
			return model.Error(ctx, msg, http.StatusBadRequest)
		}
```